vyper/compiler/phases.py

```diff
@@ -145,9 +146,33 @@ def vyper_module(self):
         _, ast = self._generate_ast
         return ast
 
+    @cached_property
+    def _resolve_imports(self):
+        vyper_module = copy.deepcopy(self.vyper_module)
+        with self.input_bundle.search_path(Path(vyper_module.resolved_path).parent):
+            return vyper_module, resolve_imports(vyper_module, self.input_bundle)
+
+    @cached_property
+    def resolved_imports(self):
+        imports = self._resolve_imports[1]
+
+        expected = self.expected_integrity_sum
+
+        if expected is not None and imports.integrity_sum != expected:
+            # warn for now. strict/relaxed mode was considered but it costs
+            # interface and testing complexity to add another feature flag.
+            vyper_warn(
+                f"Mismatched integrity sum! Expected {expected}"
+                f" but got {imports.integrity_sum}."
+                " (This likely indicates a corrupted archive)"
+            )
+
+        return imports
+
     @cached_property
     def _annotate(self) -> tuple[natspec.NatspecOutput, vy_ast.Module]:
-        module = generate_annotated_ast(self.vyper_module, self.input_bundle)
+        module = self._resolve_imports[0]
+        analyze_module(module)
```

Comment:
i don't like the name `analyze_module` - like even the property is called `annotate`

can we change to smth which would better imply the annotation process?

Reply:
it is analysis though, it does things besides just annotation

Reply:
i don't associate side-effects with analysis. like do analysis and, based on it, perform annotation.

 ..but it's not very important